docs/setup/settings.asciidoc

```diff
@@ -504,6 +504,87 @@ deprecation warning at startup. This setting cannot end in a slash (`/`).
 proxy sitting in front of it. This determines whether HTTP compression may be used for responses, based on the request `Referer` header.
 This setting may not be used when <<server-compression, `server.compression.enabled`>> is set to `false`. *Default: `none`*
 
+
+a| [[server-securityResponseHeaders-strictTransportSecurity]]
+----
+server.securityResponseHeaders:
+  strictTransportSecurity:
+----
+| Controls whether the https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security[`Strict-Transport-Security`]
+header is used in all responses to the client from the {kib} server. Set this to `null` to disable it. *Default:* enabled
+
+a|
+----
+server.securityResponseHeaders:
+  strictTransportSecurity.maxAge:
+----
+| Specifies the `max-age` directive of the
+https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security[`Strict-Transport-Security`] response header. This is
+rounded down to the nearest second. *Default:* `'1Y'` (31536000 seconds)
```

Comment:
As discussed, the settings here are a bit inconsistent right now with the transport policy header being controlled using individual settings whereas permissions policy having to be set manually as the raw header value.

Would be good to make that consistent. Maybe the easiest thing would be to let users set the raw header value themselves so that we don't have to keep updating it manually when the spec changes.

Reply:
Agreed, done in c358373!